src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -77,21 +83,38 @@ public function getProperties($class, array $context = [])
             return;
         }
 
-        $reflectionProperties = $reflectionClass->getProperties();
+        $propertyFlags = 0;
+        $methodFlags = 0;
+
+        if ($this->accessFlags & self::ALLOW_PUBLIC) {
```

Comment:
would it make sense to have this configurable via $context?

Reply:
I want to avoid context as much as possible as it make metadatas harder to cache for an use case that it's IMO very rare. And if you need different behaviors you can always declare multiple services.